pkg/ccl/backupccl/backup_test.go

```diff
@@ -3616,6 +3616,174 @@ func TestBackupRestoreSequence(t *testing.T) {
+		}
+	})
+
+	t.Run("test restoring table then sequence should remove ownership dependency",
+		func(t *testing.T) {
+			// When just the table is restored by itself, the ownership dependency is
+			// removed as the sequence doesn't exist. When the sequence is restored
+			// after that, it requires the `skip_missing_sequence_owners` flag as
+			// the table isn't being restored with it, and when provided, the sequence
+			// shouldn't have an owner.
+			tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{ServerArgs: args})
+			defer tc.Stopper().Stop(context.Background())
+
+			newDB := sqlutils.MakeSQLRunner(tc.Conns[0])
+			kvDB := tc.Server(0).DB()
+			newDB.Exec(t, `CREATE DATABASE d`)
+			newDB.Exec(t, `USE d`)
+			newDB.ExpectErr(t, `pq: cannot restore sequence "seq" without referenced owner table`,
+				`RESTORE TABLE seq FROM $1`, backupLoc)
+
+			newDB.ExpectErr(t, `pq: cannot restore table "t" without referenced sequence`,
+				`RESTORE TABLE t FROM $1`, backupLoc)
+			newDB.Exec(t, `RESTORE TABLE t FROM $1 WITH skip_missing_sequence_owners`, backupLoc)
+
+			tableDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "t")
+
+			numOwnedSequences := len(tableDesc.GetColumns()[0].OwnsSequenceIds)
+			if numOwnedSequences != 0 {
+				t.Fatalf("expected restored table to own 0 sequences. got: %v", numOwnedSequences)
+			}
+
+			newDB.ExpectErr(t, `pq: cannot restore sequence "seq" without referenced owner table`,
+				`RESTORE TABLE seq FROM $1`, backupLoc)
+			newDB.Exec(t, `RESTORE TABLE seq FROM $1 WITH skip_missing_sequence_owners`, backupLoc)
+
+			seqDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "seq")
+			if seqDesc.SequenceOpts.HasOwner() {
+				t.Fatal("unexpected sequence owner after restore")
+			}
+		})
+
+	t.Run("test restoring database should preserve ownership dependency", func(t *testing.T) {
+		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{ServerArgs: args})
+		defer tc.Stopper().Stop(context.Background())
+
+		newDB := sqlutils.MakeSQLRunner(tc.Conns[0])
+		kvDB := tc.Server(0).DB()
+
+		newDB.Exec(t, `RESTORE DATABASE d FROM $1`, backupLoc)
+
+		tableDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "t")
+		seqDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "seq")
+
+		if !seqDesc.SequenceOpts.HasOwner() ||
+			seqDesc.SequenceOpts.SequenceOwner.OwnerTableID != tableDesc.ID ||
+			seqDesc.SequenceOpts.SequenceOwner.OwnerColumnID != tableDesc.GetColumns()[0].ID {
+			t.Fatal("unexpected/no sequence owner after restore")
+		}
+
+		if len(tableDesc.GetColumns()[0].OwnsSequenceIds) != 1 ||
+			tableDesc.GetColumns()[0].OwnsSequenceIds[0] != seqDesc.ID {
+			t.Fatal("unexpected/no sequence owned by column")
+		}
+	})
+
+	t.Run("test restoring two databases removes cross-database ownership dependency",
```

Comment:
This is really counterintuitive, but I guess any sort of cross database reference handling is strange.

Reply:
I'm not sure that it's super odd since that during the first restore you'd be explicitly removing the dependency.

I would like to see some tests restoring both databases at the same time and ensuring that the dependency is still there.